### PR TITLE
Compiler plugin read improvements

### PR DESCRIPTION
**Pull request by koperagen**
Just a series of bug fixes and tests, no new fancy features

### CODE REVIEW

File: plugins/kotlin-dataframe/src/org/jetbrains/kotlinx/dataframe/plugin/impl/api/flatten.kt [hunk 1]
```diff
@@ -14,7 +14,7 @@ import org.jetbrains.kotlinx.dataframe.plugin.impl.toPluginDataFrameSchema
 class FlattenDefault : AbstractSchemaModificationInterpreter() {
     val Arguments.receiver by dataFrame()
     val Arguments.keepParentNameForColumns: Boolean by arg(defaultValue = Present(false))
-    val Arguments.separator: String by arg(defaultValue = Present("."))
+    val Arguments.separator: String by arg(defaultValue = Present("_"))
```
Comment (zaleslaw) [inline plugins/kotlin-dataframe/src/org/jetbrains/kotlinx/dataframe/plugin/impl/api/flatten.kt:17]: is this reflects our new convention about separator in column names?

**Review by zaleslaw — APPROVED** [review]

File: plugins/kotlin-dataframe/src/org/jetbrains/kotlinx/dataframe/plugin/impl/api/flatten.kt (hunk 1 above)
Comment (koperagen) [reply plugins/kotlin-dataframe/src/org/jetbrains/kotlinx/dataframe/plugin/impl/api/flatten.kt:17]: Yes :)

**Review by Jolanrensen — APPROVED** [review]

File: plugins/kotlin-dataframe/src/org/jetbrains/kotlinx/dataframe/plugin/impl/api/read.kt [hunk 2]
```diff
@@ -74,17 +79,14 @@ internal class ReadJson0 : AbstractInterpreter<PluginDataFrameSchema>() {
...
+    return when (val source = tryResolveFile(resolutionPath, path)) {
+        is ResolutionDirFile -> DataFrame.readJson(source.file)
+        is UrlOrAbsolutePath -> DataFrame.readJson(source.path)
     }
-    return df
 }
 
-private fun resolveFile(resolutionPath: String?, path: String): File? {
-    return  resolutionPath?.let {
+private fun tryResolveFile(resolutionPath: String?, path: String): DataSource {
```
[the remark this answers is not in the document]
Comment (koperagen) [reply plugins/kotlin-dataframe/src/org/jetbrains/kotlinx/dataframe/plugin/impl/api/read.kt:88]: No, it's plugin-specific. It's to handle relative path arguments like read("file.csv") when project configuration provides resolutionPath - directory where relative files are expected to be
Usual DataFrame.read implicitly relies on working directory of JVM to resolve relative files, which for compiler process is somewhere outside project directory

[the remark this answers is not in the document]
Comment (Jolanrensen) [reply plugins/kotlin-dataframe/src/org/jetbrains/kotlinx/dataframe/plugin/impl/api/read.kt:88]: aah I see :)

**Pull request merged**